pages/partials/attachFilePanel.ejs

```diff
@@ -0,0 +1,99 @@
+<div class="card mb-4" id="attach-file-panel">
+  <div class="card-header bg-secondary text-white">Attached files</div>
+  <table class="table table-sm">
+    <tbody>
+    <% if (file_list.length == 0) { %>
+    <tr><td><i>No attached files</i></td></tr>
+    <% } else { %>
+      <% file_list.forEach(function(file, iFile) { %>
+      <tr>
+        <td style="word-break:break-all;">
+          <i class="far fa-file-alt"></i>
+          <a class="attached-file" target="_blank" href="<%= urlPrefix %>/assessment_instance/<%= assessment_instance.id %>/file/<%= file.id %>/<%= file.display_filename %>">
+            <%= file.display_filename %>
+          </a>
+        </td>
+        <% if (assessment_instance.open) { %>
+        <td style="width:1%; text-align:right;">
+          <% if (file.type == 'student_upload') { %>
+          <a id="attachFileDeleteButton<%= iFile %>"
+             tabindex="0" class="btn btn-xs btn-secondary attachFileDeleteButton" role="button"
+             data-toggle="popover" data-container="body"
+             data-html="true" data-placement="auto" title="Confirm delete"
+             data-content="<%= include('attachFileDeleteForm',
+                           {id: 'attachFileDeleteButton' + iFile,
+                           variant: (typeof variant != 'undefined') ? variant : null,
+                           file}); %>"
+             data-trigger="manual" onclick="$(this).popover('show')">
+            <i class="far fa-trash-alt"></i>
+          </a>
+          <% } %>
+        </td>
+        <% } %>
+      </tr>
+      <% }); %>
+    <% } %>
+    </tbody>
+  </table>
+
+  <div class="card-footer">
+    <% if (!assessment_instance.open) { %>
+    <p class="small mb-0">
+      Attachments can't be added or deleted because the assessment is closed.
+    </p>
+    <% } else { /* assessment_instance is open */ %>
+    <!-- UPLOAD FILE -->
+    <button class="btn btn-xs btn-secondary" type="button" data-toggle="collapse" data-target="#attachFileCollapse" aria-expanded="false" aria-controls="attachFileCollapse">
+      Attach a file <i class="far fa-caret-square-down"></i>
+    </button>
+    <div class="collapse" id="attachFileCollapse">
+      <form class="attach-file-form" name="attach-file-form" method="POST" enctype="multipart/form-data" class="mb-3">
+        <p class="small mt-3">
+          Attached files will be saved here for your reference.
```

Comment:
This should mention the maximum allowable filesize, if any.

Reply:
done